src/gda/backend_gda.hpp

```diff
@@ -536,6 +541,32 @@ class GDABackend : public Backend {
    * @brief initialize function table for MLNX direct verbs support
    */
   int mlx5_dv_dl_init();
+
+  /**
+   * @brief open mlx5 dv lib
+   */
+  static void* mlx5_dv_dlopen();
+
+  /**
+   * @brief structures holding the function pointers to the direct verbs functionality
+   * of each network driver.
+   */
+  ionicdv_funcs_t ionic_dv;
```

Comment:
```suggestion
  extern ionicdv_funcs_t ionic_dv;
```
This is a header, globals should be extern. Define `ionic_dv` in `backend_gda.cpp`.

Actually, I think this can be local to `src/gda/ionic/backend_gda_ionic.cpp`, I don't think it's used anywhere else.
```C++
  static ionicdv_funcs_t ionic_dv;
```

Reply:
Its not a global, its a private member of backend_gda class; suggesion to make it a local to the ionic/backend_gda_ionic would work for ionic but not for bnxt without further refactoring